### ISSUE TITLE
A question on the design of cgroup.cpu_cfs_util

### ISSUE THREAD

**Issue by rayx** [autotest/tp-qemu#1622]
I have looked at the test code, and have a doubt on the test's design. 

If I understand it correctly, the test creates VMs and put the vcpu threads of these VMs into cgroups so that they can use as much as the max capability of all host cpus. Then it starts a number of while-loops in each VM to use up that capability. Then it checks if the cpu time used by guest is close to that used by host. 

My question is about the setup. The test sets up cgroups to allow VMs use up all host cpu cores (implementation details: assuming host has 40 cores, the test creates VMs with total number of 80 vcpus, and allow each vcpu thread to use 0.5 core in term of cpu time. So in total it allows these VMs to use all cpu time on the host). From my understanding, when one verifies cgroup setup, one should check the actual resource usage doesn't exceed the limit. In this case, however, since the limit is the entire host cpu time, I wonder how it can be checked? Or let me put it this way, if we remove the cgroup setup in this test, the test would still have same result, doesn't it? (I think so, thought I didn't verify it).

In my opinion, the test should be changed to set up cgroups to allow all VMs can use, say, 50% of host cpu time. Then run workload (eg., the while-loop) in VMs and check it doesn't exceed the limit by using the data in /proc/stat.

**Comment by xiangchunfu** [comment]
When booting a VM, avacodo-vt framework used only 1/2 all host cpu cores. It doesn't appear vcpu overcommit for single VM.

**Comment by rayx** [comment]
@xiangchunfu Sorry, I don't get what you meant. How does it relate to my original post? Can you elaborate it a bit?

**Comment by xiangchunfu** [comment]
@rayx I tested this case, As the test allow VMs use twice host cpu cores. so it is a special case.  In general,  avacodo-vt framework just allow VM to use 1/2 all host cpu cores. so please ignore my comments above.  In my understanding, Currently the test code doesn't check single vcpu if exceed the limit, It only check the entire host cpu limit.  You want to change this design so that can confirm per vcpu  if exceed limit, right?

**Comment by rayx** [comment]
@xiangchunfu No, that's not what I meant. I meant the way how cgroup is tested is fundamentally flawed in cpu_cfs_util test. Let me put it in another way. Suppose you have a process (it can be any process on host, not necessarily QEMU) and you want to test cpu.cfg_period_us and cpu.cfs_quota_us work with this process. Will you set up these two parameters to allow the process use 100% of all cpu cores on the host? You won't, because if so how would you check the setup actually works?

I think the test makes the same mistake.
